### ISSUE TITLE
Xtext cannot find DSL source when breakpoint is hit if its path has a space

### ISSUE THREAD

**Issue by swgpearson** [eclipse/xtext-core#1371]
When running traced generated code an Xtext DSL cannot be debugged if its workspace path contains a space.
See https://www.eclipse.org/forums/index.php/m/1820722/

**Comment by cdietrich** [comment]
i am not sure what the standard says regarding the FileName. and thus if the problem is in Xtext or JDT (AbstractSourceLookupParticipant/JavaSourceLookupParticipant)

I also dont know where is uris in the traces are used elsewhere and thus if the creation or the reading/writing to class files (TraceAsSmapInstaller/TraceAsPrimarySourceInstaller) needs to be adopted

**Comment by cdietrich** [comment]
@szarnekow what do you think?

**Comment by cdietrich** [comment]
ping @szarnekow

**Comment by szarnekow** [comment]
To be honest, I don't have a clue what we should really install into the smap files in these cases.
What I'm certain about: If we write encoded URIs, we should also read these as encoded URIs (e.g. decode them).

**Comment by cdietrich** [comment]
well there is two reads:
- us writing stuff into the class file
- jdt reading the source attr in the class file
we have only the first one under control

so you mean we should decode before writing into the class files?

**Comment by cdietrich** [comment]
@rytina could you please create a bug?

**Comment by szarnekow** [comment]
Yes, the path in the class file should not contain encoded  char sequences but the decoded values, e.g. spaces instead of percent20
